compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -210,13 +210,11 @@ impl<Prov> Allocation<Prov> {
         let slice: Cow<'a, [u8]> = slice.into();
         let size = Size::from_bytes(slice.len());
         let align_usize: usize = align.bytes().try_into().unwrap();
-        let count_align = ((slice.len() / align_usize) + 1)*align_usize;
-        
-        let mut vec_align: Vec<u8> = Vec::with_capacity(count_align);
-        vec_align.resize(count_align, 0);
-        // TODO avoid excess initialization
-        let (buf, _len, _capacity) = vec_align.into_raw_parts();
-        vec_align = unsafe {Vec::from_raw_parts(buf as *mut u8, size.bytes_usize(), size.bytes_usize())};
+        let layout = std::alloc::Layout::from_size_align(slice.len(), align_usize).unwrap();
+        let vec_align = unsafe {
+            let buf = std::alloc::alloc(layout);
+            Vec::from_raw_parts(buf as *mut u8, size.bytes_usize(), size.bytes_usize())
```

Comment:
Nit: You constructed `size` from `slice.len()`, maybe just use `slice.len()` here rather than roundtripping through the variable precision type.